### ISSUE TITLE
Signal Viewer no output

### ISSUE THREAD

**Issue by exekutive** [collin80/SavvyCAN#223]
The signal viewer doesn't seem to work. The value column stays blank.

Also the combo box should be sorted or organised somehow. It can be difficult to find the parameter you want in a very long list.

<img width="1087" alt="Screen Shot 2019-12-01 at 14 20 49" src="https://user-images.githubusercontent.com/4755215/69921395-16bd6a80-1446-11ea-8290-61cf6559d6d6.png">

<img width="1196" alt="Screen Shot 2019-12-01 at 13 35 27" src="https://user-images.githubusercontent.com/4755215/69921452-93504900-1446-11ea-8608-442c2004a45a.png">

**Comment by exekutive** [comment]
update: I closed and reopened signal view and the value is now there. But it's frozen and doesn't live update.

**Comment by collin80** [comment]
Hmmm, well, I just checked in Linux and adding signals does seem to work the first try and they do update. So, given this and your other issue I'm starting to suspect that there is a problem with the OSX build. I'll have to check there and find out what's going on with OSX.

**Comment by exekutive** [comment]
Do you have a Mac box to test on? I can put a little bit of time into compiling this thing.

**Comment by collin80** [comment]
Kind of... I have a virtualbox image of MacOS that I use for testing. Shh.... The actual binary you downloaded was automatically generated by a CI step so I didn't actually compile it. I just did very quick testing to see that it would run. I was naively assuming that code that runs fine on Linux and compiles on MacOS would work the same. But, unfortunately that's not always the case.

**Comment by exekutive** [comment]
yeah nothing is ever truly cross platform. Let's see if I have the proper toolchain and how many dependencies I'm missing ...

**Comment by collin80** [comment]
Hmm... I fired up my virtual copy of MacOS 10.14 and re-downloaded SavvyCAN from the website so I could be sure I had what everyone else is running. Then I tried loading a Nissan Leaf DBC file and a capture from a 2013 Leaf. I could graph things properly and they'd show up in view and properly scaled. I could load the capture in the playback window then use the signal viewer to select a couple of signals and playing back the log properly shows the signals to be updating and displaying data. I tried the UDS window too but without any capture hardware hooked up it doesn't seem to do a lot. But, it didn't appear to get stuck either. So, at this point I'm striking out on reproducing any of the issues. But, I have heard of other people having issues with MacOS and SavvyCAN. I just haven't found the reason for that. 

Maybe it would help to use real hardware. But, I really am not a fan of Apple or MacOS so I really don't particularly want to purchase a Mac. It's bad enough that I have a hand-me-down iPad. Still, I might have to break down and get something I can run MacOS 10.14 or 10.15 on.
